src/bun.js/bindings/JSBuffer.cpp

```diff
@@ -2283,6 +2283,20 @@ JSC_DEFINE_HOST_FUNCTION(constructJSBuffer, (JSC::JSGlobalObject * lexicalGlobal
     if (distinguishingArg.isAnyInt()) {
         throwScope.release();
         return JSBuffer__bufferFromLength(lexicalGlobalObject, distinguishingArg.asAnyInt());
+    } else if (distinguishingArg.isDouble()) {
```

Comment:
```suggestion
    } else if (distinguishingArg.isNumber()) {
```

Reply:
wouldnt any int thats not double be caught by .isAnyInt() ?

Reply:
```
inline bool JSValue::isDouble() const
{
    return isNumber() && !isInt32();
}
```